src/OSPSuite.UI/Binders/CurveBinder.cs

```diff
@@ -319,34 +319,33 @@ private void refreshData()
 
          // works for different base grids
          _dataTable.BeginLoadData();
-         foreach (var baseValue in baseGrid.Values)
+         baseGrid.Values.Each((baseValue, baseIndex) =>
          {
             try
             {
-               double x = xDimension.BaseUnitValueToUnitValue(xUnit, xData.GetValue(baseValue));
-               double y = yDimension.BaseUnitValueToUnitValue(yUnit, yData.GetValue(baseValue));
+               double x = xDimension.BaseUnitValueToUnitValue(xUnit, ValueInBaseUnit(xData, baseGrid, baseIndex));
+               double y = yDimension.BaseUnitValueToUnitValue(yUnit, ValueInBaseUnit(yData, baseGrid, baseIndex));
 
                if (!isValidXValue(x) || !IsValidYValue(y))
-                  continue;
+                  return;
```

Comment:
What is the reason for this behavior change?
Before, invalid values were **skipped**, but all following (valid) values were displayed.
Now, refresh stops at the first invalid value???

Reply:
oh, that's because the ```foreach``` was converted to a linq style ```.Each``` where the lamda action is exited early.

The meaning is the same. The action continues with the next item from the enumerable.

Reply:
oh, I see, thx!